git-cli/staging.md

```diff
@@ -30,6 +30,8 @@ We do this by promoting these changes files to the **staging area**.
 
 🔑 **Staging area** is the space to prepare (stage) the changes that will be in the next commit.
 
+This is useful because we don't always want to commit all of our changes at the same time.
```

Comment:
I think we should maybe give some examples here, but maybe that's going too deep here... But we haven't really motivated the reason for a staging area at all...

Reply:
One way of motivating the staging area at this stage ( pun intended 😛 ) might be to get them to think about what Git would be like if there were _no_ Staging Area. They could discuss a scenario where they modify a bunch of files in the working tree, and then decide they want to group some of the changes as separate commits. Something like this, what do you think?

Reply:
I think it's a good motivation, but I'm not sure "having separate commits" is something the trainees are motivated by yet? They aren't at a point where they want to make things easy to review, or easy to track, or easy to revert... (If we fix that, so they _are_ motivated by this, maybe it works :D)

Maybe we could try something like "If you need to put a password in a file to make something work, and you want to commit everything except for that one file with the password" is something more within the imagination?

Reply:
How about returning to the staging area later on  - in a later git session?

We can't load them up with everything at once - they won't absorb it - but getting a few dings in code review on adding their vscode settings usually provokes some discussion. Can we pick that up in JS2?

Reply:
I don't think the staging area is a huge leap at this point. Even if the motivation isn't perfect here I still think it's worth mentioning.